src/tests/ftest/launch.py

```diff
@@ -20,164 +20,465 @@
+    print("Running {}".format(cmd))
+    start_time = int(time.time())
+    subprocess.call(cmd, shell=True)
+    end_time = int(time.time())
+    return end_time - start_time
+
+
+def get_build_environment():
+    """Obtain DAOS build environment variables from the .build_vars.json file.
+
+    Returns:
+        dict: a dictionary of DAOS build environment variable names and values
+
     """
-    (base, _ext) = os.path.splitext(path)
-    return base + ".yaml"
+    build_vars_file = os.path.join(
+        os.path.dirname(os.path.realpath(__file__)),
+        "../../../.build_vars.json")
+    with open(build_vars_file) as vars_file:
+        return json.load(vars_file)
+
+
+def set_test_environment():
+    """Set up the test environment.
+
+    Returns:
+        None
 
-def printhelp():
     """
-    Print a list of test categories.
+    build_vars = get_build_environment()
+    bin_dir = build_vars['PREFIX'] + '/bin'
+    sbin_dir = build_vars['PREFIX'] + '/sbin'
+    path = os.environ.get('PATH')
+    os.environ['PATH'] = bin_dir + ':' + sbin_dir + ':' + path
+    os.environ['DAOS_SINGLETON_CLI'] = "1"
+    os.environ['CRT_CTX_SHARE_ADDR'] = "1"
+    os.environ['CRT_ATTACH_INFO_PATH'] = build_vars['PREFIX'] + '/tmp'
+
+
+def get_yaml_data(yaml_file):
+    """Get the contents of a yaml file as a dictionary.
+
+    Args:
+        yaml_file (str): yaml file to read
+
+    Raises:
+        Exception: if an error is encountered reading the yaml file
+
+    Returns:
+        dict: the contents of the yaml file
+
     """
-    print("\n")
-    print("Tests are launched by specifying a category.  For example:\n")
-    print("\tbadconnect --run pool connect tests that pass NULL ptrs, etc.  ")
-    print("\tbadevict --run pool client evict tests that pass "
-          "NULL ptrs, etc.  ")
-    print("\tbadexclude --run pool target exclude tests that "
-          "pass NULL ptrs, etc.  ")
-    print("\tbadparam --run tests that pass NULL ptrs, etc.  ")
-    print("\tbadquery --run pool query tests that pass NULL ptrs, etc.  ")
-    print("\tmulticreate --run tests that create multiple pools at once ")
-    print("\tmultitarget --run tests that create pools over multiple servers")
-    print("\tpool --run all pool related tests")
-    print("\tpoolconnect --run all pool connection related tests")
-    print("\tpooldestroy --run all pool destroy related tests")
-    print("\tpoolevict --run all client pool eviction related tests")
-    print("\tpoolinfo --run all pool info retrieval related tests")
-    print("\tquick --run tests that complete quickly, with minimal resources ")
-    print("\n")
-    print("You can also specify the sparse flag -s to limit output to "
-          "pass/fail.")
-    print("Example command: launch.py -s pool")
-    print("\n")
-    exit()
-
-def run_test(_file, use_tags=True):
+    yaml_data = {}
+    if os.path.isfile(yaml_file):
+        with open(yaml_file, "r") as file:
```

Comment:
(pylint-redefined-builtin) Redefining built-in 'file'

Reply:
Fixed